tmf/org.eclipse.tracecompass.tmf.analysis.xml.core/src/org/eclipse/tracecompass/internal/tmf/analysis/xml/core/output/XmlDataProviderManager.java

```diff
@@ -321,4 +333,76 @@ public void refreshDataProviderFactories() {
         fTimeGraphFactories.clear();
         fXYFactories.clear();
     }
+
+    /**
+     * Get the XML data provider for a trace, provider id and XML
+     * {@link OutputType}
+     *
+     * @param trace
+     *            the queried trace
+     * @param id
+     *            the queried ID
+     * @param types
+     *            the data provider types
+     * @return the provider if an XML containing the ID exists and applies to
+     *         the trace, else null
+     */
+    @SuppressWarnings("unchecked")
+    public <@Nullable P extends ITmfTreeDataProvider<? extends ITmfTreeDataModel>> P getXmlProvider(ITmfTrace trace, String id, Set<OutputType> types) {
```

Comment:
I think this needs to be:
public @Nullable <P extends ...

(that's Nullable with a capital N, thank you github and shout-out to @nullable)

Reply:
I'll update that.